src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -788,8 +847,114 @@ static void Main()
                   ILocalReferenceExpression: num (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'num')
                   InConversion: null
                   OutConversion: null
+  NextVariables(0)
 ";
             VerifyOperationTreeForTest<ForEachStatementSyntax>(source, expectedOperationTree);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForEachLoopStatement_WithDeconstructDeclaration()
+        {
+            string source = @"
+class X
+{
+    public static void M((int, int)[] x)
+    {
+        /*<bind>*/foreach (var (a, b) in x)
+        {
+        }/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+IForEachLoopStatement (LoopKind.ForEach) (OperationKind.LoopStatement) (Syntax: 'foreach (va ... }')
+  Locals: Local_1: System.Int32 a
```

Comment:
The formatting for this needs to be fixed to print the first local on a new line, but the dumper code for dumping locals is shared between variables list and block locals and caused large number of tests to get affected. Hence I reverted the formating change and will file a separate issue to track fixing it.

Reply:
Filed https://github.com/dotnet/roslyn/issues/21827